Geometry/TrackerGeometryBuilder/src/TrackerGeometry.cc

```diff
@@ -129,6 +129,8 @@ void TrackerGeometry::addDetUnitId(DetId p){
 }
 
 void TrackerGeometry::addDet(GeomDet const * p) {
+  // set index
+  const_cast<GeomDet *>(p)->setGdetIndex(theDets.size());
```

Comment:
static analyzer is not happy with (yet another) const_cast in this file.
Why are they even here, should the method signature be changed instead?